grafana/grafana
# Datasource Onboarding: Prevent flickering of onboarding page after first load

Showing the DS onboarding page is currently dependant on the `state.dataSources.hasFetched` state property, which once set to `true` after loading the list of datasource instances the first time, does never get set back to `false`. this causes a flicker (as in showing the onboarding page briefly) when navigating to a new dashboard or explore because - while `hasFetched` is still set to `true` - we need to refetch datasources every times that page renders to also decide wether to show that page in the first place or not:

https://user-images.githubusercontent.com/1170767/219000513-4a008e98-074c-4835-accf-83281e069dfa.mov

(see #63609 for more details)

this PR fixes that by slightly refactoring the dasources state by renaming  `isLoadingDataSources` to `isLoadingDataSourcePlugins`, and use `isLoadingDataSources` to indicate if the datasource list is currently being loaded

Fixes #63609


**How to test this**:
with `datasourceOnboarding` feature toggle on, make sure you have no data source setup (is using devenv comment out everything in `devenv/datasources.yaml` and remove grafana db):

- go to new dashboard (or explore), the onboarding page should appear
- create a datasource
- try to create a new dashboard (or navigate to explore), the onboarding page should not briefly show up.
